docs/development/building-from-sources.md

```diff
@@ -2,85 +2,119 @@
+4).
+```
+
+If running `make` deterministically stops at one point,
+increasing the maximum RAM usage available to the docker container might help
+[This is different from the physical RAM capacity inside the system]. Ideally,
+at least 3 GB of RAM should be available to the docker container to build
+Pyodide smoothly. These settings can be changed via Docker Preferences (See
+[here](https://stackoverflow.com/questions/44533319/how-to-assign-more-memory-to-docker-container)).
+
+You can edit the files in your source checkout on your host machine, and then
+repeatedly run `make` inside the Docker environment to test your changes.
+
+## Using `make`
 
 Make sure the prerequisites for
 [emsdk](https://github.com/emscripten-core/emsdk) are installed. Pyodide will
 build a custom, patched version of emsdk, so there is no need to build it
 yourself prior.
 
+You would need Python 3.9.5 to run the build scripts. To make sure that the
+correct Python is used during build it is recommended to use a [Python virtual
+environment](https://packaging.python.org/guides/installing-using-pip-and-virtual-environments/#creating-a-virtual-environment),
+
+```{tabbed} Linux
+
 Additional build prerequisites are:
 
 - A working native compiler toolchain, enough to build
   [CPython](https://devguide.python.org/setup/#linux).
-- A native Python 3.9 to run the build scripts.
 - CMake
-- PyYAML
 - FreeType 2 development libraries to compile Matplotlib.
-- Cython to compile SciPy
-- SWIG to compile NLopt
 - gfortran (GNU Fortran 95 compiler)
 - [f2c](http://www.netlib.org/f2c/)
 - [ccache](https://ccache.samba.org) (optional) _highly_ recommended for much faster rebuilds.
+- (optional) SWIG to compile NLopt
 
-You can install the python dependencies from the requirement file at the root of pyodide folder:
-`pip install -r requirements.txt`
+```
 
-On Mac, you will also need:
+```{tabbed} MacOS
+
+On MacOS, you would need:
 
 - [Homebrew](https://brew.sh/) for installing dependencies
 - System libraries in the root directory (
   `sudo installer -pkg /Library/Developer/CommandLineTools/Packages/macOS_SDK_headers_for_macOS_10.14.pkg -target /`
   should do it, see https://github.com/pyenv/pyenv/issues/1219#issuecomment-428305417)
-- coreutils for md5sum and other essential Unix utilities (`brew install coreutils`)
+- coreutils for md5sum and other essential Unix utilities (`brew install coreutils`).
 - cmake (`brew install cmake`)
-- Cython to compile SciPy (`brew install cython`)
-- SWIG to compile NLopt (`brew install swig`)
 - pkg-config (`brew install pkg-config`)
 - openssl (`brew install openssl`)
 - gfortran (`brew cask install gfortran`)
 - f2c: Install wget (`brew install wget`), and then run the buildf2c script from
   the root directory (`sudo ./tools/buildf2c`)
+- It is also recommended to install the GNU patch (`brew install gpatch`), and
+  GNU sed (`brew install gnu-sed`) and re-define them temporarily as patch and
+  sed.
```

Comment:
Is there a link to someone explaining how exactly to do this? It seems like someone could need to do some googling to figure out how to apply this advice in practice, might be nice to save them the trouble with a link.

Reply:
I added a link to https://formulae.brew.sh/formula/gnu-sed which explains how to do it. Though I haven't tried building on MacOS (aside from the CI), so this is mostly speculation, but I think it should fix those recurrent issues with patch/sed.